### PR TITLE
Add sha256 field in package file

### PR DESCRIPTION
**Pull request by valentinmanea**
Added the sha256 field in PackageFile class

Fixes: #1142

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/models/PackageFile.java
Comment (jmini) [inline src/main/java/org/gitlab4j/api/models/PackageFile.java:1]: Please add a value in the example file `package-files.json`. This way we are sure the serialisation/deserialisation code works (tested in the `TestGitLabApiBeans` unit test)

**Comment by jmini** [comment]
I have rebased the source branch of this PR (`6.x_add_sha256_field_in_package_file` in the `valentinmanea` fork) on top of `main` of this repo:

```
git rebase --onto origin/main origin/6.x
git push valentinmanea HEAD:6.x_add_sha256_field_in_package_file --force
```

Be aware that your local branch `6.x_add_sha256_field_in_package_file` has diverged with the one on the GitHub.

If you need to do one more changes on this branch:
I recommend to use `git fetch` and `git reset --hard origin/6.x_add_sha256_field_in_package_file` (assuming your local `origin` is pointing to the `valentinmanea` fork on GitHub) before doing any local modifications instead of `git pull`.

I have changed the target branch of this PR to be `main`

**Review by jmini — APPROVED** [review]

**Comment by valentinmanea** [comment]
@jmini Can you merge this ?

**Comment by jmini** [comment]
I have been fighting CI, because of the flakiness of some integrations tests. But now it is good.

**Comment by jmini** [comment]
I have merged your change to `main` and `6.x`. 🎉 

Thank you very much for this contribution.

**Pull request merged**